contracts/exchangeIssuance/DEXAdapter.sol

```diff
@@ -21,147 +21,176 @@ import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
 
+import { ICurveCalculator } from "../interfaces/external/ICurveCalculator.sol";
+import { ICurveAddressProvider } from "../interfaces/external/ICurveAddressProvider.sol";
+import { ICurvePoolRegistry } from "../interfaces/external/ICurvePoolRegistry.sol";
+import { ICurvePool } from "../interfaces/external/ICurvePool.sol";
 import { ISwapRouter} from "../interfaces/external/ISwapRouter.sol";
+import { IWETH } from "../interfaces/IWETH.sol";
 import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
 
 
 
+
 /**
  * @title DEXAdapter
  * @author Index Coop
  *
  * Adapter to execute swaps on different DEXes
  *
  */
-abstract contract DEXAdapter {
+library DEXAdapter {
     using SafeERC20 for IERC20;
     using PreciseUnitMath for uint256;
     using SafeMath for uint256;
 
     /* ============ Constants ============= */
 
     uint256 constant private MAX_UINT256 = type(uint256).max;
-    uint24 public constant POOL_FEE = 3000;
+    address public constant ETH_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    // This is the margin in wei we add on top of stEth amounts to avoid errors. 
+    // This is needed both in aave as well as curve integration 
+    // TODO: Review to find better solution
```

Comment:
@ncitron : This is the rounding issue I mentioned in slack. Which I also encountered when doing a curve exact-output-swap (which lead to revertion due to the swap underbuying by 1 or 2 wei).

Unless you veto it, I'd like to just keep this as is now and remove the TODO comment.

Reply:
Yeah I agree that this is a fine concession to make. Nobody is going to miss 2 wei. I think the problem has something to do with astETH rebases